### PR TITLE
DCS-1400 Fix role definitions and remove dependency on mockk

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,6 +14,7 @@ ARG BUILD_NUMBER
 ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
 
 RUN apt-get update && \
+    apt-get -y upgrade && \
```
Comment (andrewrlee) [inline Dockerfile:17]: fix trivy alert - ensure image packages are updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt [hunk 2]
```diff
@@ -51,7 +50,6 @@ class ArrivalsService(
       arrival.copy(prisonNumber = null)
   }
 
-  @PreAuthorize("hasRole('BOOKING_CREATE') and hasAuthority('SCOPE_write')")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:54]: removed this as we enforce this at the controller level

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt [hunk 3]
```diff
@@ -112,8 +112,7 @@ class PrisonApiClient(@Qualifier("prisonApiWebClient") private val webClient: We
       .uri("/api/users/me/caseLoads")
       .retrieve()
       .bodyToMono(typeReference<List<UserCaseLoad>>())
-      .onErrorResume(WebClientResponseException::class.java) { emptyWhenNotFound(it) }
-      .block()
+      .block() ?: emptyList()
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt:115]: fix warning as possibly returns null

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt [hunk 4]
```diff
@@ -78,14 +78,6 @@ class PrisonService(
       )
       .offenderNo
 
-  companion object {
-    private val numbers = '0'..'9'
-    private val letters = 'A'..'Z'
-
-    private operator fun CharRange.invoke(n: Int) = this.shuffled().take(n).joinToString("")
-    private fun randomPrisonNumber() = letters(1) + numbers(4) + letters(2)
-  }
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt:87]: not needed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ArrivalsResource.kt [hunk 5]
```diff
@@ -131,11 +131,11 @@ class ArrivalsResource(
     @PathVariable moveId: String,
   ): Arrival = arrivalsService.getArrival(moveId)
 
-  @PreAuthorize("hasRole('BOOKING_CREATE') and hasAuthority('SCOPE_write')")
+  @PreAuthorize("hasRole('ROLE_BOOKING_CREATE') and hasRole('ROLE_TRANSFER_PRISONER') and hasAuthority('SCOPE_write')")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ArrivalsResource.kt:134]: this requires TRANSFER_PRISONER if the arrival is court return

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**